esmvaltool/diag_scripts/crem/ww09_esmvaltool.py

```diff
@@ -0,0 +1,575 @@
+            else:
+                pointers[key] = selection[0]["filename"]
+                vartags.append('V_' + var)
+                missing_snow = False
+                break
+
+        if missing_snow:
+            missing_vars.append(ww_vars_plus[0] + " or " + ww_vars_plus[1])
+
+        for fn in pointers:
+            climofiles.append(','.join(fn))
+
+        # check if all variables are available
+
+        if missing_vars:
+            printlist = ', '.join(missing_vars)
+            logger.error("error: the following variables are not "
+                         "available: %s", printlist)
+            raise Exception('Variables missing (see log file for details).')
+
+        # calculate CREM
+
+        (crem_pd, r_crem_pd) = crem_calc(pointers)
+
+        crems[i] = crem_pd
+        i = i + 1
+
+    logger.info("------------------------------------")
+    logger.info(crems)
+    logger.info("------------------------------------")
+
+    # plot results
+
+    if cfg['write_plots']:
+        oname = os.path.join(
+            cfg['plot_dir'],
+            'ww09_metric_multimodel.' + cfg['output_file_type'],
+        )
+        logger.debug("Plotting results to %s", oname)
+
+        plt.figure()
+        ypos = np.arange(nummod)
+        plt.barh(ypos, crems, align='center')
+        plt.yticks(ypos, models)
+        plt.xlabel('Cloud Regime Error Metric')
+
+        # draw observational uncertainties (dashed red line)
+        plt.plot([0.96, 0.96], [-0.5, nummod - 0.5], 'r--')
+
+        plt.savefig(oname, bbox_inches='tight')
+
+        # add meta data to plot (for reporting)
+
+#        basetags = 'TO BE DONE'
+#
+#        ESMValMD("both",
+#            oname,
+#            basetags + ['DM_global', 'PT_bar'] + modeltags + vartags,
+#            'Cloud Regime Error Metric (CREM) following Williams and Webb '
+#            '(2009, Clim. Dyn.).',
+#            '#ID_ww09_crem',
+#            ','.join(climofiles), 'ww09_ESMValTool.py', 'A_will_ke')
+
+    if cfg['write_netcdf']:
+        oname = os.path.join(cfg['work_dir'], 'ww09_metric_multimodel.nc')
+        logger.debug("Saving results to %s", oname)
+        # convert strings
+        str_out = np.array(models, dtype=object)
+        # open a new netCDF file for writing
+        ncfile = Dataset(oname, 'w')
+        # create the x dimension
+        ncfile.createDimension('n', nummod)
+        # create the variable
+        # first argument is name of variable, second is datatype, third is
+        # a tuple with the name(s) of dimension(s)
+        data = ncfile.createVariable('crem', np.dtype('float32').char, ('n'))
+        mods = ncfile.createVariable('model', str, ('n'))
+        # write data to variable
+        data[:] = crems
+        mods[:] = str_out
+        # close the file
+        ncfile.close()
+
+
+# Reading and Regridding functions (scroll down for main program)
+
+
+def regrid(a_in, x_in, y_in, x_out, y_out, fixmdis=True, x_cyclic=0.0):
```

Comment:
Would it be possible to use the [regrid](https://esmvaltool.readthedocs.io/en/version2_development/codedoc2/esmvaltool.preprocessor.html#esmvaltool.preprocessor.regrid) function from the preprocessor? Or do you really need the custom implementation.

Reply:
This regridding has some special features such as cyclic treatment of the longitudes and optional replacement of interpolated values with exact values if source and target lat/lon are sufficiently close. I think in the mid- to long-term this should be done by the preprocessor (as listed in the header of the script under "to do") but I would leave it as it is for now for the sake of being able to reproduce the v1.1.0 results.